src/js/plugins/plugin.search.view.js

```diff
@@ -0,0 +1,317 @@
+  }
+
+  buildToolbarSearch() {
+    const toolbarSearch = document.createElement('span');
+    toolbarSearch.classList.add('BRtoolbarSection', 'BRtoolbarSectionSearch');
+    toolbarSearch.innerHTML = `
+      <form class="BRbooksearch desktop">
+        <input type="search" name="query" class="BRsearchInput" val="" placeholder="Search inside"/>
+        <button type="submit" class="BRsearchSubmit">
+          <img src="${this.br.imagesBaseURL}icon_search_button.svg" />
+        </button>
+      </form>
+    `;
+    return toolbarSearch;
+  }
+
+  renderPins(matches) {
+    matches.forEach((match) => {
+      const queryString = match.text;
+      const pageIndex = this.br.leafNumToIndex(match.par[0].page);
+      const pageNumber = this.br.getPageNum(pageIndex);
+      const uiStringSearch = "Search result"; // i18n
+      const uiStringPage = "Page"; // i18n
+
+      const percentThrough = this.br.constructor.util.cssPercentage(pageIndex, this.br.getNumLeafs() - 1);
+
+      const queryStringWithB = queryString.replace(this.matcher, '<b>$1</b>');
+
+      let queryStringWithBTruncated = '';
+
+      if (queryString.length > 100) {
+        queryStringWithBTruncated = queryString
+          .replace(/^(.{100}[^\s]*).*/, "$1")
+          .replace(this.matcher, '<b>$1</b>')
+                + '...';
+      }
+
+      // draw marker
+      const markerTopValue = -this.br.refs.$brContainer.height();
+      $('<div>')
+        .addClass('BRsearch')
+        .css({
+          top: `${markerTopValue}px`,
+          left: percentThrough,
+        })
+        .attr('title', uiStringSearch)
+        .append(`
+          <div class="BRquery">
+            <div>${queryStringWithBTruncated || queryStringWithB}</div>
+            <div>${uiStringPage} ${pageNumber}</div>
+          </div>
+        `)
+        .data({ pageIndex })
+        .appendTo(this.br.$('.BRnavline'))
+        .hover(
+          (event) => {
+            // remove from other markers then turn on just for this
+            // XXX should be done when nav slider moves
+            const marker = event.currentTarget;
+            const tooltip = marker.querySelector('.BRquery');
+            const tooltipOffset = tooltip.getBoundingClientRect();
+            const targetOffset = marker.getBoundingClientRect();
+            const boxSizeAdjust = parseInt(getComputedStyle(tooltip).paddingLeft) * 2;
+            if (tooltipOffset.x - boxSizeAdjust < 0) {
+              tooltip.style.setProperty('transform', `translateX(-${targetOffset.left - boxSizeAdjust}px)`);
+            }
+            $('.BRsearch,.BRchapter').removeClass('front');
+            $(event.target).addClass('front');
+          },
+          (event) => $(event.target).removeClass('front'))
+        .click(function (event) {
+          // closures are nested and deep, using an arrow function breaks references.
+          // Todo: update to arrow function & clean up closures
+          // to remove `bind` dependency
```

Comment:
ok sounds good.

Reply:
This was a comment from the existing search plugin code. I didn't quite understand it, but figured I could leave it until we come back and remove jQuery.